CONTRIBUTING.md

```diff
@@ -17,15 +17,63 @@ try to keep the codebase consistent and clean.
 
 ## Getting Started
 
-* Make sure you have a GitHub account.
+* Make sure you have a GitHub account and solid ability to use `git`.
 * Fork the repository on GitHub.
-* Create a topic branch from master. Please avoid working directly on the ```master``` branch.
+* Create a topic branch from master. Please avoid working directly on the `master` branch.
 * Make commits of logical units.
-* Check for coding style issues with ```git diff master..mybranch | ./sys/clang-format-diff.py -p1``` and be sure to follow the CODINGSTYLE (more on this in [DEVELOPERS.md](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md)).
+* Check for coding style issues with:
+
+      git diff master..mybranch | ./sys/clang-format-diff.py -p1
+
+  and be sure to follow the CODINGSTYLE (more on this in [DEVELOPERS.md][]).
 * Submit the Pull Request(PR) on Github.
 * When relevant, write a test in [test/](test).
 
+## Rebasing onto updated master
+
+Every so often, your PR will lag behind `master` and get conflicts.
+
+To "update" your branch `my-awesome-feature`, you *rebase* it onto
```

Comment:
Actually it is also possible to merge master into your branch, which is IMHO easier/faster at times, because you just need to resolve the conflicts one time instead of, possibly, N times, one for each of your commit.

Reply:
Well yes, merging-vs-rebasing is a hot topic in itself, there're both pros & cons to each...

But from the perspective of project maintainer — I find that *merging master into feature branch* produces absolutely disgusting history graphs. It takes way more effort to make any sense of such history (should anything need go wrong and need reverting, cherry-picking, cross-merging, etc) — compared to the neat linear rebased-feature-branches.

Reply:
Yes, however 99% of the time we do not merge multiple commits in master, but we do squash+merge. This means that even if you have your super clean history of commits on top of master (and I assure you that 99% of the times you don't), everything will be reduced to one single commit.

For this reason using rebasing or merging, when updating an old PR, doesn't really matter, because at the end of the day everything will be unified in one single commit. So, knowing that, a `git merge origin/master` in my PR-branch is what I usually do as I need to resolve conflicts just one time and the end result is the same (plus for me that means I don't have to use `--force` ever, because as I have permission to write on multiple branches, a small mistake somewhere could make me force-push wrong stuff on some branches).

Reply:
Both reasonings make sense to me, but i'm more with @ulidtko , it requires more work to rebase+cherrypick.. than just merge master .. which only looks fine in github if you squash+rebase from the github web. otherwise it's a mess and it becomes much more confusing to reference commits by index and merge commits appear empty while they group other ones.. so imho we can say that:

* merge is fine if the whole PR will end up being 1 squashed commit and you let others to commit or your PR will take a lot to be finished
* rebase if you know what you are the only commiter in there and you have proper git skills to not loose your changes

Additionally we can also explain how

Reply:
OK so mentioning `merge master` here will require mentioning `squash` -- which requires going more in-depth into interactive rebase. That's rather big detour off the flow. You sure want to put this into `CONTRIBUTING.md` ? I can do it no problem, it's just a bit too much IMO.

Reply:
merging master into the pr doesn't require using squash at all. Never used `squash`. The way we work and we have always done in the last X years (don't remember exactly how much) is that we squash+merge PRs into master. So even if your PR has 10 commits, they will appear as one single commit in master. We even have squash+merge as the default button for merging PRs.